src/aws/osml/model_runner/inference/http_detector.py

```diff
@@ -0,0 +1,109 @@
+import logging
+from io import BufferedReader
+from json import JSONDecodeError
+from typing import Optional
+
+import geojson
+import urllib3
+from aws_embedded_metrics.logger.metrics_logger import MetricsLogger
+from aws_embedded_metrics.metric_scope import metric_scope
+from aws_embedded_metrics.unit import Unit
+from geojson import FeatureCollection
+
+from aws.osml.model_runner.api import ModelInvokeMode
+from aws.osml.model_runner.app_config import MetricLabels, ServiceConfig
+from aws.osml.model_runner.common import Timer
+
+from .detector import Detector
+from .endpoint_builder import FeatureEndpointBuilder
+from .feature_utils import create_mock_feature_collection
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPDetector(Detector):
+    def __init__(self, endpoint: str) -> None:
+        """
+        A HTTP model endpoint invoking object, intended to query sagemaker endpoints.
+
+        :param endpoint: str = the full URL to invoke the model
+
+        :return: None
+        """
+        self.http_pool = urllib3.PoolManager(cert_reqs="CERT_NONE")
+        super().__init__(endpoint=endpoint)
+
+    @property
+    def mode(self) -> ModelInvokeMode:
+        return ModelInvokeMode.HTTP_ENDPOINT
+
+    @metric_scope
+    def find_features(self, payload: BufferedReader, metrics: MetricsLogger) -> FeatureCollection:
+        """
+        Query the established endpoint mode to find features based on a payload
+
+        :param payload: BufferedReader = the BufferedReader object that holds the
+                                    data that will be sent to the feature generator
+        :param metrics: MetricsLogger = the metrics logger object to capture the log data on the system
+
+        :return: FeatureCollection = a feature collection containing the center point of a tile
+        """
+        retry_count = 0
+        logger.info("Invoking HTTP Endpoint: {}".format(self.endpoint))
+        if isinstance(metrics, MetricsLogger):
+            metrics.set_dimensions()
+            metrics.put_dimensions({"HTTPModelEndpoint": self.endpoint})
```

Comment:
This metric should really to match the dimension used by the SMDetector. Consider adding another parameter to the HTTP Endpoint to give it a name instead of just adding the full URL here.

Reply:
Added default name of "http" and the ability to pass in a custom name to give the endpoint if needed.